ReactMarkdownをアップデート

ocha98 opened the pull request:
ReactMarkdownをアップデート

## 変更内容
ReactMarkdownを更新しました。新しいバージョンに対応できるように修正しました。

GFM記法に対応し表やリンクに対応しました。

また、数式のレンダリングをrehype-katexに変更しました。これにより、 #123  を解決しました。

ocha98 committed:
[Update] Markdown view

`Markdown.tsx`
  GFMの記法に対応するためにremark-gfmを追加
  数式のレンダリングをremark-math + rehype-katexに変更

  `rehypeInlineCodeProperty`関数
    インラインコードとコードブロックを親要素がpreかどうかで識別
    インラインコードの場合はclassNameに`inline`を追加
    コードブロックの場合は親要素のpreを無効化

  `bootstrapTable`関数
    生成されるテーブルにスタイルを付与するための関数

  ReactMarkdownをアップデートしたためそれに応じた変更を加えました

`MarkdownStyle.module.css`
  hタグの要素に下線を引くために作成

その他、パッケージをアップデートしました

makutamoto commented:
![image](https://github.com/makutamoto/mojacoder/assets/34860673/190a5065-1a9c-4f12-a975-824baca42991)
いつもありがとうございます！ところでこのエラーはそちらで再現していますか？

ocha98 commented:
再現できてないですね。
一旦全消去し再度リポジトリをクローンしてビルドしてみましたがこちらではエラーは出てこなかったです（＞＜；；）

ocha98 commented:
プルリクのバージョンとは違いますが、Next.jsのバージョンを14まで上げると私の環境でも似たようなエラーが発生しました。

修正してみます。

![image](https://github.com/makutamoto/mojacoder/assets/48848793/c1989298-1566-4e02-b0ec-7c043eb18aec)

makutamoto commented:
よろしくお願いします

ocha98 committed:
[Update] Next.js14

next/linkが<a>タグを生成するようになったので不要な<a>を消去
Bootstrapの<Nav.Link>を囲うものに対してはlegacyBehaviorをtrueにする

next/imageがaltを必須にしたので、altを追加

`Username.tsx`のレイアウトがアップデートにより崩れたので修正

ocha98 commented:
修正しました

エラーの原因は`<a><a></a></a>`のような不正なHTML構造になっていたことでした

## 変更点
Next.js14へアップグレードしました

Next.js13からnext/linkの仕様が変わっていたので対応しました
next/imageのalt属性が必須になっていたのでaltを付与しました

一部レイアウトが崩れてしまったので修正しました

Node.js 20.10.0で動作確認しました

makutamoto committed:
Next jsをアップデートしたことによる制約に対応

makutamoto commented:
デプロイされました．確認お願いします

ocha98 commented:
上手くいってないようです。マークダウンのテーブル表記が使えるようになるのですがそれが確認できなかったです。

今回破壊的変更が多くて本当にすみません。

makutamoto commented:
![image](https://github.com/makutamoto/mojacoder/assets/34860673/fb6f050e-9550-4b1f-bde1-350ab5065460)
AmplifyのCIがNext.jsのバージョンを上げたときにぶっ壊れていて，フロントエンドの変更が一切反映されていないことがわかりました．治った．
